File: gen_scripts/ruby/m00.function.rb
#!/usr/bin/env ruby

# create method (function)
def addNums (*numbers)
   sum = 0                          # initialize starting sum value
   numbers.each {|num| sum += num } # iterate and sum up nums
   sum                              # return value
end

numbers = [5, 2, 4, 3, 6]
# output numbers before function
puts "The numbers to be added are #{numbers.join(", ")}."

# call the method (function)
result = addNums(*numbers)

# output results
puts "The result of their summation is: #{result}."

File: compiled_lang/go/src/m00.function.go
package main

import "fmt"

func addNums(numbers ...int) int {
	sum := 0
	for _, num := range numbers {
		sum += num
	}
	return sum
}

func main() {
	fmt.Println("The numbers to be added are 5, 2, 4, 3, 6.")

	result := addNums(5, 2, 4, 3, 6)
	fmt.Printf("The result of their summation is: %d.\n", result)
}
